packages/login/src/common/LoginBackgroundWrapper.tsx

```diff
@@ -57,29 +57,36 @@ export interface IProps {
 
 export function usePersistentCountryBackground() {
   const theme = useTheme()
-  const countryBackground: NonNullable<
-    ReturnType<typeof selectCountryBackground>
-  > = JSON.parse(
-    localStorage.getItem('country-background') ??
-      `{"backgroundColor" : "${(theme as ITheme).colors.backgroundPrimary}"}`
-  )
+  const [countryBackground, setCountryBackground] = React.useState({
+    backgroundColor: `${(theme as ITheme).colors.backgroundPrimary}`,
+    backgroundImage: '',
+    imageFit: ''
+  })
   const [offlineBackground, setOfflineBackground] =
     React.useState(countryBackground)
+  React.useEffect(() => {
+    storage.getItem('country-background').then((res) => {
+      res ?? setCountryBackground(JSON.parse(res))
+    })
```

Comment:
I wonder how this is working :thinking: I read it as "if `res` is undefined then run `setCountryBackground(JSON.parse(res))`". Don't we want it to be the other way around like "if `res` is not undefined then run `setCountryBackground(JSON.parse(res))`"

Reply:
but if this was not working, then the countryBackground shouldn't show up(which I assume is showing up properly) so that got me confused